diff --git a/CMakeLists.txt b/CMakeLists.txt
index 61e1118..0dc92d2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,19 +46,19 @@ endif()
 
 if(USEGPU) 
 	if (CUDAToolkit_VERSION_MAJOR STREQUAL "10")
-		set(CUDA_ARCH_LIST "50;52;53;60;61;62;70;72;75")
+		set(CUDA_ARCH_LIST "35;37;50;72;75")
 	elseif (CUDAToolkit_VERSION_MAJOR STREQUAL "11")
 		if(CUDAToolkit_VERSION_MINOR STREQUAL "0")
-			set(CUDA_ARCH_LIST "50;52;53;60;61;62;70;72;75")
+			set(CUDA_ARCH_LIST "50;72;75")
 		elseif (CUDAToolkit_VERSION_MINOR STREQUAL "1" OR CUDAToolkit_VERSION_MINOR STREQUAL "2")
-			set(CUDA_ARCH_LIST "50;52;53;60;61;62;70;72;75;80;86")
+			set(CUDA_ARCH_LIST "50;72;75;80;86")
 		else () # for now, we assume CUDA 11.2+ supports all these archs.
-			set(CUDA_ARCH_LIST "50;52;53;60;61;62;70;72;75;80;86")
+			set(CUDA_ARCH_LIST "50;72;75;80;86")
 		endif()
 	elseif (CUDAToolkit_VERSION_MAJOR STREQUAL "12")
-		set(CUDA_ARCH_LIST "50;52;53;60;61;62;70;72;75;80;86;89;90")
+		set(CUDA_ARCH_LIST "52;60;75;80;86;89;90")
 	else() # some old CUDA version (<10)
-		set(CUDA_ARCH_LIST "50") # 5.0 is the oldest non-deprecated CC as of 2024-09-06
+		set(CUDA_ARCH_LIST "50")
 	endif()
 endif()
 
